### PR TITLE
EPMDJ-6475 Fix calculation of risks for lambda and methods containing…

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**
… lambda

### CODE REVIEW

File: admin-part/src/main/kotlin/jvm/Ext.kt [hunk 1]
```diff
@@ -24,20 +24,55 @@ import java.io.*
...
+ * Bootstrap methods have a fixed set of arguments, the second of them is the index of the lambda method
+ * in the constant pool.
+ * For more info: https://blogs.oracle.com/javamagazine/post/behind-the-scenes-how-do-lambda-expressions-really-work-in-java
+ */
+internal fun JavaClass.getParsedBootstrapMethods() = attributes
+    .filterIsInstance<BootstrapMethods>()
+    .firstOrNull()
+    ?.bootstrapMethods
+    ?.map { bootstrapMethod ->
+        bootstrapMethod.bootstrapArguments[1]
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/jvm/Ext.kt:66]:
```suggestion
        val refLamda = 1
        bootstrapMethod.bootstrapArguments[refLamba]
```
or smth another name

File: admin-part/src/main/kotlin/jvm/Ext.kt [hunk 2]
```diff
@@ -24,20 +24,55 @@ import java.io.*
...
+
+/**
+ * For each lambda creates a bootstrap method, which make references back into the main constant pool.
+ * Bootstrap methods have a fixed set of arguments, the second of them is the index of the lambda method
+ * in the constant pool.
+ * For more info: https://blogs.oracle.com/javamagazine/post/behind-the-scenes-how-do-lambda-expressions-really-work-in-java
+ */
+internal fun JavaClass.getParsedBootstrapMethods() = attributes
+    .filterIsInstance<BootstrapMethods>()
+    .firstOrNull()
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/jvm/Ext.kt:63]: maybe
```suggestion
    .first
```

File: admin-part/src/main/kotlin/Methods.kt [hunk 3]
```diff
@@ -90,6 +103,10 @@ internal fun List<Method>.diff(otherMethods: List<Method>): DiffMethods = if (an
     } else DiffMethods(new = this)
 } else DiffMethods(deleted = otherMethods)
 
+private fun MutableList<Method>.addMethod(value: Method) {
+    if ("lambda" !in value.name) add(value)
```
Comment (Mazel-Tovr) [inline admin-part/src/main/kotlin/Methods.kt:107]: Can we make "lambda" constant field ?
We also use it in admin-part/src/main/kotlin/jvm/Ext.kt file

File: admin-part/src/main/kotlin/jvm/Ext.kt [hunk 4]
```diff
@@ -24,20 +24,57 @@ import java.io.*
...
+    }.map { (classIndex, nameAndTypeIndex) ->
+        val classFullName = (getConstant(classIndex.nameIndex) as ConstantUtf8).bytes
+        val methodName = (getConstant(nameAndTypeIndex.nameIndex) as ConstantUtf8).bytes
+        val signature = (getConstant(nameAndTypeIndex.signatureIndex) as ConstantUtf8).bytes
+        signature(classFullName, methodName, signature)
+    }
+
+private fun JavaClass.getConstant(it: Int) = constantPool.getConstant(it)
+
+private fun JavaClass.arrayOfBootstrapMethods() = attributes
```
Comment (Mazel-Tovr) [inline admin-part/src/main/kotlin/jvm/Ext.kt:78]: mb remove prefix "arrayOf"?
```suggestion
private fun JavaClass.classBootstrapMethods() = attributes
```

**Review by Mazel-Tovr — APPROVED** [review]

**Pull request merged**